src/mpid/ch4/generic/am/mpidig_am_msg.h

```diff
@@ -109,10 +109,16 @@ MPL_STATIC_INLINE_PREFIX void MPIDIG_recv_copy(void *in_data, MPIR_Request * rre
         MPIR_STATUS_SET_COUNT(rreq->status, 0);
     } else if (p->recv_type == MPIDIG_RECV_DATATYPE) {
         MPI_Aint actual_unpack_bytes;
-        MPIR_Typerep_unpack(in_data, in_data_sz,
-                            MPIDIG_REQUEST(rreq, buffer),
-                            MPIDIG_REQUEST(rreq, count),
-                            MPIDIG_REQUEST(rreq, datatype), 0, &actual_unpack_bytes);
+        /* RECV in MPI allows NULL recv buffer with non-zero data_sz, here we skip the attempt of
+         * unpacking and just count the recv counters */
+        if (MPIDIG_REQUEST(rreq, buffer)) {
```

Comment:
I'm pretty sure this is going to break legitimate usage of `MPI_BOTTOM`.

Reply:
Yeah, that too.

Reply:
Ah, that's right. I guess we can fix this the other way by not using NULL buffer on recv for the offending test.

Reply:
> Ah, that's right. I guess we can fix this the other way by not using NULL buffer on recv for the offending test.

https://github.com/pmodels/mpich/blob/b2011442b9bc3bed150ad7d2c62e25c2ee849feb/test/mpi/datatype/hindexed_zeros.c#L50

Seems like the test is technically correct usage. I don't think we should change it just to avoid the warning. `count` may not be zero, but the total number of receive bytes is zero. Can we capture this in typerep and return instead of calling down to yaksa?

Reply:
@yfguo Could you paste the warning/failure here? Why it was not triggered before?